src/main.js

```diff
@@ -557,8 +557,9 @@ define([
   Peaks.prototype.setSource = function(options, callback) {
     var self = this;
 
-    if (!options.mediaUrl) {
-      callback(new Error('peaks.setSource(): options must contain a mediaUrl'));
+    if ((options.mediaElement || this.options.mediaElement) && !options.mediaUrl) {
```

Comment:
`mediaElement` can only be set in `Peaks.init()`. It shouldn't be a valid option in `setSource()`.